src/controllers/vendor/vendorsController.js

```diff
@@ -0,0 +1,288 @@
+dotenv.config();
+
+const app = express();
+app.use(express.urlencoded({ extended: false }));
+app.use(i18nextMiddleware.handle(i18next));
+i18next
+  .use(Backend)
+  .use(i18nextMiddleware.LanguageDetector)
+  .init({
+    backend: {
+      loadPath: '././src/locales/{{lng}}/{{ns}}.json',
+    },
+    fallbackLng: 'en',
+    preload: ['en', 'fr'],
+  });
+export const getAllVendors = async (req, res) => {
+  try {
+    const { authenticatedUser } = req;
+    if (!authenticatedUser) {
+      return res.status(403).json({
+        success: false,
+        message: req.t('unauthorized_msg'),
+      });
+    }
+    const vendor = await vendors.findAll();
+    res.status(200).json({
+      success: true,
+      message: req.t('getAllVendors_200_msg'),
+      response: vendor,
+    });
+  } catch (error) {
+    console.log(error.message);
+    res.status(500).json({
+      success: false,
+      message: req.t('getAllVendors_500_msg'),
+      Error: error.message,
+    });
+  }
+};
+
+export const registerVendor = async (req, res) => {
+  try {
+    const { authenticatedUser } = req;
+    if (!authenticatedUser) {
+      return res.status(403).json({
+        success: false,
+        message: req.t('unauthorized_msg'),
+      });
+    }
+    const { error } = validateVendorRegistration(req.body);
+    if (error) {
+      console.log(error);
+      return res.status(400).json({
+        success: false,
+        message: req.t('registerVendor_400_msg'),
+        Error: error.details[0].message,
+      });
+    }
+    const vendorEmail = await vendors.findOne({
+      where: {
+        email: req.body.email,
+      },
+    });
+    if (vendorEmail) {
+      return res.status(409).json({ success: false, message: req.t('registerVendor_409_msg') });
+    }
+    const hashedPassword = await bcrypt.hash(req.body.password, 10);
+    const newVendor = await vendors.create({
+      fullName: req.body.fullName,
+      email: req.body.email,
+      password: hashedPassword,
+      phoneNumber: req.body.phone,
+      businessName: req.body.businessName,
+      businessAddress: req.body.businessAddress,
+      businessPhoneNumber: req.body.businessPhone,
+      businessEmail: req.body.businessEmail,
+      businessWebsite: req.body.businessWebsite,
+      businessDescription: req.body.businessDescription,
+      businessLogo: req.body.businessLogo,
+      productCategories: req.body.productCategories,
+      paymentMethods: req.body.paymentMethods,
+      status: req.body.status,
+    });
+    if (
+      !req.body.fullName
+      || !req.body.email
+      || !req.body.password
+      || !req.body.phoneNumber
+      || !req.body.businessName
+      || !req.body.businessAddress
+      || !req.body.businessPhoneNumber
+      || !req.body.businessEmail
+      || !req.body.businessWebsite
+      || !req.body.businessDescription
+      || !req.body.businessLogo
+      || !req.body.productCategories
+      || !req.body.paymentMethods
+      || !req.body.status
+    ) {
```

Comment:
You can remove this because you already validated them..

Reply:
Sure, DONE!!!